targets/psa_switch/psa_switch.cpp

```diff
@@ -313,6 +301,16 @@ PsaSwitch::ingress_thread() {
                     ingress_port);
 
     phv = packet->get_phv();
+
+    /* Ingress cloning and resubmitting work on the packet before parsing.
+       `buffer_state` contains the `data_size` field which tracks how many
+       bytes are parsed by the parser ("lifted" into p4 headers). Here, we
+       track the buffer_state prior to parsing so that we can put it back
+       for packets that are cloned or resubmitted, same as in simple_switch.cpp
+
+       TODO update this along with simple_switch.cpp's analogous section */
+    const Packet::buffer_state_t packet_in_state = packet->save_buffer_state();
```

Comment:
you can use `auto packet_in_state = ...`

Reply:
sure! will make the change - but just for transparency (and since I'm quite new to c++) whats the purpose of favoring `auto` over the explicit type?

Reply:
I'll refer you to "Effective Modern C++" by Scott Meyers
In this specific case, it is mostly a matter of personal preference and uniformity. I do prefer not repeating the type when it can be avoided. If anyone ever needs to rename / change the return type, it will mean less changes.